modules/export_pipeline/export_to_aws.tf

```diff
@@ -0,0 +1,109 @@
+# Create a service account used by the function and Eventarc trigger
+resource "google_service_account" "export_to_aws" {
+  account_id   = "gcf-aws-export-sa"
+  display_name = "Used By Export To AWS Cloud Function"
+}
+
+# Grant permissions needed for the cloud function to run and receive event triggers.
+resource "google_project_iam_member" "export_to_aws_invoking" {
+  project    = data.google_project.project.project_id
+  role       = "roles/run.invoker"
+  member     = "serviceAccount:${google_service_account.export_to_aws.email}"
+  depends_on = [google_project_iam_member.gcs_pubsub_publishing]
+}
+
+resource "google_project_iam_member" "export_to_aws_receiving" {
+  project    = data.google_project.project.project_id
+  role       = "roles/eventarc.eventReceiver"
+  member     = "serviceAccount:${google_service_account.export_to_aws.email}"
+  depends_on = [google_project_iam_member.export_to_aws_invoking]
+}
+
+resource "google_project_iam_member" "export_to_aws_artifactregistry_reader" {
+  project    = data.google_project.project.project_id
+  role       = "roles/artifactregistry.reader"
+  member     = "serviceAccount:${google_service_account.export_to_aws.email}"
+  depends_on = [google_project_iam_member.export_to_aws_receiving]
+}
+
+# Grant access to Secret Manager
+resource "google_project_iam_member" "export_to_aws_secret_manager_access" {
```

Comment:
Did you create the secret manager manually?
I think it'd be good to create it in here as well. Ideally we'd be limiting access to either 
- a secret manager which only contains keys needed by this cloud function
- a specific sub-set of keys within a secret manager needed by this cloud function

The thing I'd want to avoid is giving this cloud function access to all keys inside a secret manager which could contain other keys it doesn't need.

Reply:
How would we hide the value of the keys if we create the secretes via terraform?

Reply:
Granting access to specific keys instead